### PR TITLE
Use `go-defaults` to set default value for file mode, use `validate` library to validate all blocks

### PR DESCRIPTION
**Pull request by lonegunmanb**
[go-defaults](https://github.com/mcuadros/go-defaults) seems no longer maintained and due to this [issue](https://github.com/mcuadros/go-defaults/issues/14), it cannot set default value for pointer type field, I have to create [my personal fork](https://github.com/lonegunmanb/go-defaults) and merge this [pr](https://github.com/mcuadros/go-defaults/pull/22), then publish a new tag [`v1.3.0`](https://github.com/lonegunmanb/go-defaults/tree/v1.3.0) based on the `v1.2.0`.

This pr also use [`github.com/go-playground/validator`](https://github.com/go-playground/validator) to validate all blocks.

### CODE REVIEW

File: pkg/fix_local_file.go [hunk 1]
```diff
@@ -33,22 +33,26 @@ func (lf *LocalFileFix) Type() string {
...
+		writeErr := afero.WriteFile(FsFactory(), path, []byte(lf.Content), fm)
 		if writeErr != nil {
 			err = multierror.Append(err, writeErr)
 		}
 	}
 
 	return err
 }
+
+func toDecimal(decimalMode fs.FileMode) (fs.FileMode, error) {
```
Comment (matt-FFFFFF) [inline pkg/fix_local_file.go:51]: should we call the input octalMode instead of decimalMode?

**Review by matt-FFFFFF — COMMENTED** [review]
Only one small comment.

I have tested this and it works with inputs of 0755 and 755 being equivalent.

**Comment by lonegunmanb** [comment]
> Only one small comment.
> 
> I have tested this and it works with inputs of 0755 and 755 being equivalent.

`755` would trigger a golangci-lint issue so I have to use `0755`.

**Comment by matt-FFFFFF** [comment]
> > Only one small comment.
> > I have tested this and it works with inputs of 0755 and 755 being equivalent.
> 
> `755` would trigger a golangci-lint issue so I have to use `0755`.

I tested both `755` and `0755` in the grept config and they are equivalent. I like this as people often forget to put the leading zero.

**Pull request merged**